tests/integration/test_model.py

```diff
@@ -1024,3 +1024,266 @@ def test_predict_with_nullable_types(c):
+    c.sql(
+        """
+        CREATE OR REPLACE MODEL my_model WITH (
+            model_class = 'GradientBoostingClassifier',
+            wrap_predict = True,
+            target_column = 'target'
+        ) AS (
+            SELECT x, y, x*y > 0 AS target
+            FROM timeseries
+            LIMIT 100
+        )
+    """
+    )
+    check_trained_model(c)
+
+    model_query = """
+        CREATE OR REPLACE MODEL my_model WITH (
+            model_class = 'LogisticRegression',
+            wrap_predict = True,
+            wrap_fit = False,
+            target_column = 'target'
+        ) AS (
+            SELECT x, y, x*y > 0 AS target
+            FROM timeseries
+        )
+        """
+    c.sql(model_query)
+    check_trained_model(c)
+
+    model_query = """
+        CREATE OR REPLACE MODEL my_model WITH (
+            model_class = 'LinearRegression',
+            target_column = 'target'
+        ) AS (
+            SELECT x, y, x*y AS target
+            FROM timeseries
+        )
+        """
+    c.sql(model_query)
+    check_trained_model(c)
+
+    c.sql(
+        """
+        CREATE OR REPLACE MODEL my_model WITH (
+            model_class = 'KMeans'
+        ) AS (
+            SELECT x, y
+            FROM timeseries
+            LIMIT 100
+        )
+    """
+    )
+    check_trained_model(c)
+
+    c.sql(
+        """
+        CREATE OR REPLACE MODEL my_model WITH (
+            model_class = 'SGDClassifier',
+            wrap_fit = True,
+            target_column = 'target',
+            fit_kwargs = ( classes = ARRAY [0, 1] )
+        ) AS (
+            SELECT x, y, x*y > 0 AS target
+            FROM timeseries
+            LIMIT 100
+        )
+    """
+    )
+    check_trained_model(c)
+
+    c.sql(
+        """
+        CREATE OR REPLACE EXPERIMENT my_exp WITH (
+        model_class = 'GradientBoostingClassifier',
+        experiment_class = 'GridSearchCV',
+        tune_parameters = (n_estimators = ARRAY [16, 32, 2],learning_rate = ARRAY [0.1,0.01,0.001],
+                           max_depth = ARRAY [3,4,5,10]),
+        target_column = 'target'
+    ) AS (
+            SELECT x, y, x*y > 0 AS target
+            FROM timeseries
+            LIMIT 100
+        )
+        """
+    )
+    check_trained_model(c, "my_exp")
+
+    c.sql(
+        """
+        CREATE OR REPLACE EXPERIMENT my_exp WITH (
+        model_class = 'GradientBoostingClassifier',
+        experiment_class = 'RandomizedSearchCV',
+        tune_parameters = (n_estimators = ARRAY [16, 32, 2],learning_rate = ARRAY [0.1,0.01,0.001],
+                           max_depth = ARRAY [3,4,5,10]),
+        target_column = 'target'
+    ) AS (
+            SELECT x, y, x*y > 0 AS target
+            FROM timeseries
+            LIMIT 100
+        )
+        """
+    )
+    check_trained_model(c, "my_exp")
+
+    c.sql(
+        """
+        CREATE MODEL IF NOT EXISTS my_model_lightgbm WITH (
+            model_class = 'LGBMClassifier',
```

Comment:
LightGBM does not seem to support `dask_cudf` DataFrames. I actually get the same error `TypeError: Implicit conversion to a NumPy array is not allowed` as from https://github.com/dask-contrib/dask-sql/issues/943